### ISSUE TITLE
[Request] Getting the serial number of the car in .animated files

### ISSUE THREAD

**Issue by HUN-Droid** [leezer3/OpenBVE#619]
### Description

It would be nice to know in the animated file what serial number the car has on the train.
It could be similar to the term _delta_ or _value_.
For example: if that car is the 1st car in the train, then this term returns 1, if this is the 2nd car, then return 2, etc. If the object isn't a train car, then returns 0, or -1.

I think, this isn't a hard development (I'm not a pro), but it will be very useful for me.

Thanks.

**Comment by leezer3** [comment]
Yep, that's trivial :)

Added the variable **CarNumber** for today's daily build- https://vps.bvecornwall.co.uk/OpenBVE/Builds/OpenBVE-2021-04-25.zip
This works as follows:
* If not part of a train, returns -1
* If part of a train, return the zero-based car number the object is attached to

**Comment by HUN-Droid** [comment]
Btw the CarIndex isn't better name for this variable?
It doesn't matter to me really, I'm just asking that if the index isn't free, maybe it expresses better that it's the index of this car. For me, CarNumber means how many cars the train has.

**Comment by leezer3** [comment]
I agree on the natural language semantics to an extent, hence my note above.

---
TLDR:
**Index** is reserved internally in the language lexiography.
Take the function *speedometer*
This also has the variant *speedometer[carIndex]*
In the internal language lexiography, a function name suffixed with **index** denotes that it is the car specific variant.
Hence, these two translate to *speedometer* and *speedometerindex*

CarIndex according to the lexiography therefore accesses the property **Car** of the car of index N.

**Comment by HUN-Droid** [comment]
I see, thank you!
